.github/workflows/test-integration.yml

```diff
@@ -0,0 +1,91 @@
+name: Integration Tests
+permissions:
+  contents: read
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - 'internal/**'
+      - 'pkg/**'
+      - 'go.mod'
+      - 'go.sum'
+      - '.env.example'
+      - 'run-test-integrations.sh'
+      - 'docker-compose.test.yml'
+      - '.github/workflows/test-integration.yml'
+  pull_request:
+    branches: ["main"]
+    paths:
+      - 'internal/**'
+      - 'pkg/**'
+      - 'go.mod'
+      - 'go.sum'
+      - '.env.example'
+      - 'run-test-integrations.sh'
+      - 'docker-compose.test.yml'
+      - '.github/workflows/test-integration.yml'
+  workflow_dispatch:  # Allow manual triggering
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    services:
+      postgres:
+        image: postgres:15-alpine
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+          - 5433:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "stable"
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download Go dependencies
+        run: go mod download
+
+      - name: Setup Docker Compose
+        uses: KengoTODA/actions-setup-docker-compose@main
```

Comment:
Follow [this tutorial](https://docs.github.com/en/actions/use-cases-and-examples/using-containerized-services/creating-postgresql-service-containers) to setup Postgres